src/main/java/com/softserve/service/ArticleService.java

```diff
@@ -1,8 +1,37 @@
 package com.softserve.service;
 
+import java.util.Optional;
+
 import org.springframework.stereotype.Service;
+import org.springframework.ui.Model;
+
+import com.softserve.model.Article;
+import com.softserve.repository.IArticleRepository;
 
 @Service
 public class ArticleService {
+
+	private IArticleRepository articleRepository;
+	
+	private SubCategoryService subCategoryService;
+	
+	public ArticleService(IArticleRepository articleRepository, SubCategoryService subCategoryService) {
```

Comment:
I would suggest using @Autowired on articleRepository and subCategoryService attributes instead of initializing through the constructor